tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlobStoreManagedLedgerOffloader.java

```diff
@@ -555,13 +555,18 @@ private BlobStoreLocation getBlobStoreLocation(Map<String, String> offloadDriver
         BlobStoreLocation bsKey = getBlobStoreLocation(offloadDriverMetadata);
         String readBucket = bsKey.getBucket(offloadDriverMetadata);
         BlobStore readBlobstore = blobStores.get(config.getBlobStoreLocation());
+        String dataContent = DataBlockUtils.dataBlockOffloadKey(ledgerId, uid);
+        String indexContent = DataBlockUtils.indexBlockOffloadKey(ledgerId, uid);
 
         CompletableFuture<Void> promise = new CompletableFuture<>();
         scheduler.chooseThread(ledgerId).submit(() -> {
             try {
-                readBlobstore.removeBlobs(readBucket,
-                    ImmutableList.of(DataBlockUtils.dataBlockOffloadKey(ledgerId, uid),
-                                     DataBlockUtils.indexBlockOffloadKey(ledgerId, uid)));
+                log.info("Remove Blob Data: " + dataContent);
+                readBlobstore.removeBlob(readBucket, dataContent);
```

Comment:
Can we use two try/catch blocks? This way if the first remove fails, we will try to delete the second (the index)
Otherwise if we fail on the data block we won't ever try to delete the index blob

Reply:
Sure.

Reply:
I have checked out the code and actually there is no way to deal with a partial failure. 
we can commit this patch in the current form